### ISSUE TITLE
Debugging SSL certificate installation script

### ISSUE THREAD

**Issue by setiah** [opensearch-project/security#1648]
Following up on [this PR](https://github.com/opensearch-project/security/pull/1637) to revert SSL script changes. 

I just tried installing SSL configurations via the updated tool in #1530 on 1.3 version and it worked fine for me. Here are the steps - 

1. Built the 1.3 OpenSearch artifact from `1.x` branch. `./gradlew clean assemble`
2. Built the 1.3 security plugin artifact from 2.0 branch. ``
3. Installed the Security plugin artifact in OpenSearch with ./bin/opensearch-plugin install file:///<security plugin artifact location>
4. Replaced the existing install_demo_configurations.sh with the updated version in #1530 .
5. Ran the tool `./plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s`

Please check the comments for logs. 

@cliu123 could you describe steps to help me reproduce the issue as you see it

**Comment by setiah** [comment]
Successful run logs

```
ubuntu@ip-172-31-29-26:~/Debug/opensearch-1.3.0-SNAPSHOT$ ./plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s
=========== OpenSearch Security Dev Configurations Installer ============

Checking if certificates exist in /home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/certs/

#############################
Step 1: Generating self-signed certificates using OpenSSL
#############################

Creating certs/ directory...
Root certificates created at /home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/certs/certs/root-ca.pem
Admin certificates created at /home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/certs/certs/kirk.pem
Node certificates created at /home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/certs/certs/esnode.pem

#############################
Step 2: Installing self-signed certificates
#############################


Certificates installed successfully! See updated settings in config/opensearch.yml

#############################
Step 3: Initialize security
#############################

OpenSearch Security will be automatically initialized.
If you like to change the runtime configuration
change the files in ../securityconfig and execute:
"/home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/plugins/opensearch-security/tools/securityadmin.sh" -cd "/home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/plugins/opensearch-security/securityconfig" -icl -key "/home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/kirk-key.pem" -cert "/home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/kirk.pem" -cacert "/home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/root-ca.pem" -nhnv
or run ./securityadmin_demo.sh
To use the Security Plugin ConfigurationGUI

#############################
Setup complete!
#############################

Single node setup successful! You can now start the cluster with bin/opensearch

To access your secured cluster open https://<hostname>:<HTTP port> and log in with admin/admin. Example
curl -k https://localhost:9200 -u admin:admin
(Ignore the SSL certificate warning because we installed self-signed certificates)

Follow the steps below to setup multi-node cluster
1. Execute this script now on all your nodes
2. Replace certificates on all other nodes with generated certificates on this node. The certificates can be found on path - /home/ubuntu/Debug/opensearch-1.3.0-SNAPSHOT/config/certs/
3. Start all nodes with bin/opensearch

For configuring language clients like Java, Python, Go, and tools like Logstash, Grafana, etc.,
please refer to the CLIENTS AND TOOLS documentation on https://opensearch.org/docs/latest

#############################
Disclaimer!
#############################

 ** Warning: Do not use on production or public reachable systems **
Self signed certificates installed via this tool are good for development or use within a trusted private network.
For any production use, we recommend using trusted CA provided certificates.
```

**Comment by CEHENKLE** [comment]
@davidlago Hey, can you take a look at this guy?  I'd like to get it building into 2.0 as soon as possible.  *nudge* :)

**Comment by cliu123** [comment]
> 3. Installed the Security plugin artifact in OpenSearch with ./bin/opensearch-plugin install file:///
4. Replaced the existing install_demo_configurations.sh with the updated version in https://github.com/opensearch-project/security/pull/1530 .

@setiah If you switch the sequence of the step 3 and step 4, that is, run the `install_demo_configuration.sh` script before installing security plugin, you will see the errors.

**Comment by setiah** [comment]
@cliu123 that's incorrect. You cannot execute the tool before installing the security plugin. The tool sets up the configurations for security plugin. It does not make sense to run the tool when the security plugin is not installed in the first place itself. 

For an end user, the security plugin comes pre-installed in the default distribution. the tool exists inside security plugin and can be found at `opensearch-security/tools/install_demo_configurations.sh`.  The user runs the setup tool and it sets up the configurations for security plugin. This is consistent with how it works with the existing tool today.

**Comment by cliu123** [comment]
> @cliu123 that's incorrect. You cannot execute the tool before installing the security plugin. The tool sets up the configurations for security plugin. It does not make sense to run the tool when the security plugin is not installed in the first place itself.
> 
> For an end user, the security plugin comes pre-installed in the default distribution. the tool exists inside security plugin and can be found at `opensearch-security/tools/install_demo_configurations.sh`. The user runs the setup tool and it sets up the configurations for security plugin. This is consistent with how it works with the existing tool today.

Oh, sorry, my bad. I was totally lost. Your steps are correct. And with those steps, not all certs get generated, and plugin installation failed. @DarshitChanpura Do you happen to have the error logs that both of us were able to reproduce?

**Comment by setiah** [comment]
>And with those steps, not all certs get generated, and plugin installation failed.

Again, plugin installation should have nothing to do with the tool. It is an independent step and should happen before running the tool.

I am guessing there is some setup issue here. It would help to see the logs (and steps) to understand this better.

**Comment by cliu123** [comment]
> > And with those steps, not all certs get generated, and plugin installation failed.
> 
> Again, plugin installation should have nothing to do with the tool. It is an independent step and should happen before running the tool.
> 
> I am guessing there is some setup issue here. It would help to see the logs (and steps) to understand this better.

I meant to say cluster bootstrap. Fixed the typo.

**Comment by DarshitChanpura** [comment]
Looks like the issue is the `-y -i -s` arguments at the end of the script install command.

<details>
<summary>If you try to run the script without the arguments, the script execution will end prematurely, resulting in the required `.pem` certs to not be generated (except `root-ca-key.pem`).</summary>

```
➜  opensearch-security $ ./tools/install_demo_configuration.sh
=========== OpenSearch Security Dev Configurations Installer ============
Generate and install self-signed certificates? [y/N] y
Cluster mode requires maybe additional setup of:
  - Virtual memory (vm.max_map_count)

Enable cluster mode? [y/N] y

Checking if certificates exist in /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/
How many days should keys be active for?
Provide subject for root certificate?
Provide subject for admin certificate?
Provide subject for node certificate?

#############################
Step 1: Generating self-signed certificates using OpenSSL
#############################

➜  opensearch-security $ 
```
</details>

<details>
<summary>But when I clean-up and run the install script again like this: `./tools/install_demo_configuration.sh -y -i -s`, the script installs correctly. </summary>

```
➜  opensearch-security $ ./tools/install_demo_configuration.sh -y -i -s
=========== OpenSearch Security Dev Configurations Installer ============

Checking if certificates exist in /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/

#############################
Step 1: Generating self-signed certificates using OpenSSL
#############################

Root certificates created at /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/certs/root-ca.pem
Admin certificates created at /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/certs/kirk.pem
Node certificates created at /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/certs/esnode.pem

#############################
Step 2: Installing self-signed certificates
#############################


Certificates installed successfully! See updated settings in config/opensearch.yml

#############################
Step 3: Initialize security
#############################

OpenSearch Security will be automatically initialized.
If you like to change the runtime configuration
change the files in ../securityconfig and execute:
"/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/plugins/opensearch-security/tools/securityadmin.sh" -cd "/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/plugins/opensearch-security/securityconfig" -icl -key "/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/kirk-key.pem" -cert "/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/kirk.pem" -cacert "/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/root-ca.pem" -nhnv
or run ./securityadmin_demo.sh
To use the Security Plugin ConfigurationGUI

#############################
Setup complete!
#############################

Single node setup successful! You can now start the cluster with bin/opensearch

To access your secured cluster open https://<hostname>:<HTTP port> and log in with admin/admin. Example
curl -k https://localhost:9200 -u admin:admin
(Ignore the SSL certificate warning because we installed self-signed certificates)

Follow the steps below to setup multi-node cluster
1. Execute this script now on all your nodes
2. Replace certificates on all other nodes with generated certificates on this node. The certificates can be found on path - /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/
3. Start all nodes with bin/opensearch

For configuring language clients like Java, Python, Go, and tools like Logstash, Grafana, etc.,
please refer to the CLIENTS AND TOOLS documentation on https://opensearch.org/docs/latest

#############################
Disclaimer!
#############################

 ** Warning: Do not use on production or public reachable systems **
Self signed certificates installed via this tool are good for development or use within a trusted private network.
For any production use, we recommend using trusted CA provided certificates.
➜  opensearch-security $ 
```
</details>


@setiah The error that @cliu123 is talking about can be reproduced by running the install script twice in succession to see the `Setup Complete` message and then attempting to start the OpenSearch server using `./bin/opensearch`.

<details>
<summary> The startup fails with `Unable to read es-node.pem cert`. Refer the stack-trace for more details. </summary>

```
org.opensearch.bootstrap.StartupException: java.lang.IllegalStateException: failed to load plugin class [org.opensearch.security.OpenSearchSecurityPlugin]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:169) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:100) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) ~[opensearch-cli-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.cli.Command.main(Command.java:101) ~[opensearch-cli-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:135) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:101) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
Caused by: java.lang.IllegalStateException: failed to load plugin class [org.opensearch.security.OpenSearchSecurityPlugin]
	at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:792) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:728) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:530) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:193) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	... 6 more
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:481) ~[?:?]
	at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:783) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:728) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:530) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:193) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	... 6 more
Caused by: org.opensearch.OpenSearchSecurityException: Error while initializing transport SSL layer from PEM: OpenSearchException[Unable to read /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/esnode.pem (/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/esnode.pem). Please make sure this files exists and is readable regarding to permissions. Property: plugins.security.ssl.transport.pemcert_filepath]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:419) ~[?:?]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]
	at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]
	at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:481) ~[?:?]
	at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:783) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:728) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:530) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:193) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	... 6 more
Caused by: org.opensearch.OpenSearchException: Unable to read /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/esnode.pem (/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/esnode.pem). Please make sure this files exists and is readable regarding to permissions. Property: plugins.security.ssl.transport.pemcert_filepath
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.checkPath(DefaultSecurityKeyStore.java:952) ~[?:?]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.resolve(DefaultSecurityKeyStore.java:237) ~[?:?]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:396) ~[?:?]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]
	at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]
	at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:481) ~[?:?]
	at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:783) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:728) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:530) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:193) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	... 6 more
uncaught exception in thread [main]
java.lang.IllegalStateException: failed to load plugin class [org.opensearch.security.OpenSearchSecurityPlugin]
Likely root cause: OpenSearchException[Unable to read /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/esnode.pem (/Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/config/certs/esnode.pem). Please make sure this files exists and is readable regarding to permissions. Property: plugins.security.ssl.transport.pemcert_filepath]
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.checkPath(DefaultSecurityKeyStore.java:952)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.resolve(DefaultSecurityKeyStore.java:237)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:396)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179)
	at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218)
	at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:783)
	at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:728)
	at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:530)
	at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:193)
	at org.opensearch.node.Node.<init>(Node.java:396)
	at org.opensearch.node.Node.<init>(Node.java:319)
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242)
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242)
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412)
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178)
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:169)
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:100)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
	at org.opensearch.cli.Command.main(Command.java:101)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:135)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:101)
For complete error details, refer to the log at /Users/dchanp/Documents/temp/opensearch-1.3.0-SNAPSHOT/logs/opensearch.log
```
</details>

**Comment by setiah** [comment]
Thanks. I ran it in interactive mode (without -y) and it worked fine. Then I ran it again with invalid inputs and I hit the error. You are seeing the failure likely because of invalid inputs provided when you run the tool in interactive mode, for e.g. when the "number of days"  is a non-positive integer or the `subject` for certificate is in invalid format. I see  the tool fails silently in that case, instead of logging the error message. I will make the change to not fail silently so user knows there is problem with the provided input.  Let me know if you see failure for valid inputs too?

**Comment by setiah** [comment]
Since the out of box experience changes are most likely pushed to 2.0 now (ongoing discussion on https://github.com/opensearch-project/opensearch-build/issues/1649), I'll push out a new PR later as per the finalized release plan. Likely we'll have the new tool under a different name, so we can deprecate the old one.

**Comment by peternied** [comment]
I am pushing this out of the v2.0.0 release since we are so close, we can accept these changes in the next minor release that is available.

**Comment by peternied** [comment]
Its been more than six months since the work that started this discussion was merged and reverted, at this point it does not look like we are making any progress so I am closing out this issue.
